activerecord/lib/active_record/relation/batches.rb

```diff
@@ -122,24 +122,97 @@ def find_in_batches(begin_at: nil, end_at: nil, batch_size: 1000, start: nil)
         end
       end
 
+      in_batches(of: batch_size, begin_at: begin_at, end_at: end_at, load: true) do |batch|
+        yield batch.to_a
+      end
+    end
+
+    # Yields ActiveRecord::Relation objects to work with a batch of records.
+    #
+    #   Person.where("age > 21").in_batches do |relation|
+    #     sleep(50)
+    #     relation.update_all(cool: true)
+    #   end
+    #
+    # If you do not provide a block to #in_batches, it will return an Enumerator
+    # which yields ActiveRecord::Relation objects for chaining with other methods:
+    #
+    #   Person.in_batches.with_index do |relation, batch|
+    #     puts "Processing relation ##{batch}"
+    #     relation.each{|relation| relation.delete_all }
+    #   end
+    #
+    # ==== Options
+    # * <tt>:of</tt> - Specifies the size of the batch. Default to 1000.
+    # * <tt>:load</tt> - Specifies if the relation should be loaded. Default to false.
+    # * <tt>:begin_at</tt> - Specifies the primary key value to start from, inclusive of the value.
+    # * <tt>:end_at</tt> - Specifies the primary key value to end at, inclusive of the value.
+    #
+    # This is especially useful if you want to work with the
+    # ActiveRecord::Relation object instead of the individual records.
+    #
+    # This is especially useful if you want multiple workers dealing with
+    # the same processing queue. You can make worker 1 handle all the records
+    # between id 0 and 10,000 and worker 2 handle from 10,000 and beyond
+    # (by setting the +:begin_at+ and +:end_at+ option on each worker).
+    #
+    #   # Let's process the next 2000 records
+    #   Person.in_batches(of: 2000, begin_at: 2000) do |relation|
+    #     relation.update_all(cool: true)
+    #   end
+    #
+    # NOTE: If you are going to iterate through each record, you should set
+    # +:load+ to true in order to reduce the number of queries. For example:
+    #
+    #   Person.in_batches(load: true).each do |batch|
+    #     batch.each(&:touch)
+    #   end
+    #
+    # NOTE: It's not possible to set the order. That is automatically set to
+    # ascending on the primary key ("id ASC") to make the batch ordering
+    # work. This also means that this method only works when the primary key is
```

Comment:
I'd change the last sentence to `Therefore the primary key must be orderable, e.g an integer or a string.`